showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,196 @@
+    try (FileWriter fileWriter = new FileWriter(tempGdchCredentialFile)) {
+      String preparedJson = converted.toPrettyString();
+      fileWriter.write(preparedJson);
+    }
+
+    // use temp location to instantiate credentials
+    initialCredentials = GdchCredentialsTestUtil.fromJson(converted, transportFactory);
+  }
+
+  /**
+   * {@link com.google.api.gax.rpc.ClientContext} will create a new {@link GdchCredentials} with an
+   * audience defaulted to the endpoint if the audience is not manually passed. This test confirms
+   * that a new credential is created from the context and can be refreshed
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testCreateClient_withGdchCredentialAndNoAudience() throws IOException {
+    context = ClientContext.create(settings);
+    stubSettings = EchoStubSettings.newBuilder(context).build();
+    client = EchoClient.create(stubSettings.createStub());
+    Credentials fromContext = context.getCredentials();
+    Credentials fromClient = initialCredentials;
+    assertNotSame(fromContext, fromClient);
+    NullPointerException expectedEx =
+        assertThrows(NullPointerException.class, () -> initialCredentials.refresh());
+    assertTrue(
+        expectedEx.getMessage().contains("Audience are not configured for GDCH service account"));
+    Exception unexpected = null;
+    try {
+      registerCredential(fromContext);
+      ((GdchCredentials) fromContext).refreshAccessToken();
+    } catch (Exception ex) {
+      unexpected = ex;
+    }
+    assertNull(unexpected);
+  }
+
+  /**
+   * Confirms creating a client with a valid audience is successful. We cannot confirm which
+   * audience is chosen (our passed audience or the endpoint) but this is confirmed in the unit
+   * tests.
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testCreateClient_withGdchCredentialWithValidAudience() throws IOException {
+    String audience = "valid-audience";
+    settings = settings.toBuilder().setGdchApiAudience(audience).build();
+    context = ClientContext.create(settings);
+    stubSettings = EchoStubSettings.newBuilder(context).build();
+    client = EchoClient.create(stubSettings.createStub());
+    Credentials fromContext = context.getCredentials();
+    assertNotSame(fromContext, initialCredentials);
+    NullPointerException thrownByClientCreds =
+        assertThrows(NullPointerException.class, () -> initialCredentials.refresh());
+    assertTrue(
+        thrownByClientCreds
+            .getMessage()
+            .contains("Audience are not configured for GDCH service account"));
+    Exception unexpected = null;
+    try {
+      registerCredential(fromContext);
+      ((GdchCredentials) fromContext).refreshAccessToken();
+    } catch (Exception ex) {
+      unexpected = ex;
+    }
+    assertNull(unexpected);
+  }
+
+  @Test
+  public void testCreateClient_withGdchCredentialWithInvalidAudience_throws() throws IOException {
```

Comment:
This and the next test cases are already tested through unit tests, I don't think we have to test them again in showcase.

Reply:
removed